fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/implementation/PreAggregateTurnOnRule.java

```diff
@@ -272,15 +276,50 @@ private boolean checkAggregations(PreAggregationContext context, PhysicalOlapSca
 
         @Override
         public Void visitPhysicalHashJoin(OptExpression optExpression, PreAggregationContext context) {
-            context.hasJoin = true;
-            context.groupings.clear();
-            context.aggregations.clear();
-            return visit(optExpression, context);
+            PhysicalHashJoinOperator hashJoinOperator = (PhysicalHashJoinOperator) optExpression.getOp();
+            OptExpression leftChild = optExpression.getInputs().get(0);
+            OptExpression rightChild = optExpression.getInputs().get(1);
+
+            ColumnRefSet leftOutputColumns = optExpression.getInputs().get(0).getOutputColumns();
+            ColumnRefSet rightOutputColumns = optExpression.getInputs().get(1).getOutputColumns();
+
+            List<BinaryPredicateOperator> eqOnPredicates = JoinPredicateUtils.getEqConj(leftOutputColumns,
+                    rightOutputColumns, Utils.extractConjuncts(hashJoinOperator.getJoinPredicate()));
+            // cross join can not do pre-aggregation
+            if (hashJoinOperator.getJoinType().isCrossJoin() || eqOnPredicates.isEmpty()) {
+                context.notPreAggregationJoin = true;
+                context.groupings.clear();
+                context.aggregations.clear();
+                return visit(optExpression, context);
+            }
+
+            // For other types of joins, only one side can turn on pre aggregation
+            ColumnRefSet usedColumns = new ColumnRefSet();
+            context.groupings.forEach(g -> usedColumns.union(g.getUsedColumns()));
+            context.aggregations.forEach(a -> usedColumns.union(a.getUsedColumns()));
```

Comment:
I think aggregation columns don't need only from one of children

Reply:
Fixed